kotlinx-coroutines-core/js/test/TestBase.kt

```diff
@@ -79,7 +81,10 @@ public actual open class TestBase actual constructor() {
     ): dynamic {
         var exCount = 0
         var ex: Throwable? = null
-        return GlobalScope.promise(block = block, context = CoroutineExceptionHandler { context, e ->
+        if (lastTestPromise != null) {
+            error("Attempt to run multiple asynchronous test within one @Test method")
```

Comment:
I think that this code looks unintuitive without additional comments surrounding it. What exactly happens when a promise is created but isn't returned? When this gets merged, `git blame` will show this PR, showing that this causes some errors  to not be reported, which makes it possible to establish the context, but I think a comment would not be too much effort but would go a long way.

Reply:
Sure thing, added a detailed explanation

Reply:
Great! Much clearer this way.